action.yml

```diff
@@ -46,11 +46,15 @@ runs:
     - name: Download hadolint and make it available in path
       shell: bash
       run: |
-        echo "::debug::Downloading Hadolint ${{ inputs.version }}"
-        mkdir ${{ github.action_path }}/bin
-        curl -L -s -o ${{ github.action_path }}/bin/hadolint \
-          "https://github.com/hadolint/hadolint/releases/download/v${{ inputs.version }}/hadolint-Linux-x86_64"
-        chmod +x ${{ github.action_path }}/bin/hadolint
+        if [ ! -f "${{ github.action_path }}/bin/hadolint" ]; then
```

Comment:
I would negate this and do the following:
- If the binary exists, run it and retrieve the version
- If it matches the version requested with the action, exit early
- If it does not match or version is not provided or the binary could not be executed, download a new version

Reply:
Oh ! Sry I didn't catch this side effect from you previous comment.
I'll update the code with requested changes

Reply:
You can try adding a test to this if you have it in you: https://github.com/jbergstroem/hadolint-gh-action/blob/main/.github/workflows/test-e2e.yml 

Optimally it covers invoking it three times in a row? first default, second with a different version and third with no version again.